integration-tests/http/__tests__/product/admin/product-category.spec.ts

```diff
@@ -0,0 +1,956 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { IProductModuleService } from "@medusajs/types"
+import { medusaIntegrationTestRunner } from "medusa-test-utils"
+import {
+  adminHeaders,
+  createAdminUser,
+} from "../../../../helpers/create-admin-user"
+
+jest.setTimeout(30000)
+
+medusaIntegrationTestRunner({
+  testSuite: ({ dbConnection, getContainer, api }) => {
+    let appContainer
+    let productCategory
+    let productCategory1
+    let productCategory2
+    let productCategoryChild
+    let productCategoryParent
+    let productCategoryChild0
+    let productCategoryChild1
+    let productCategoryChild2
+    let productCategoryChild3
+
+    let productModuleService: IProductModuleService
+
+    beforeEach(async () => {
+      appContainer = getContainer()
+
+      productModuleService = appContainer.resolve(
+        ModuleRegistrationName.PRODUCT
+      )
+
+      await createAdminUser(dbConnection, adminHeaders, appContainer)
+    })
+
+    describe("GET /admin/product-categories/:id", () => {
+      beforeEach(async () => {
+        productCategoryParent = await productModuleService.createCategory({
```

Comment:
Let's perform the creation through HTTP, so we are closer to what an end-user would do (and module changes won't break the tests if the HTTP layer remains constant)

Reply:
Good push, will do